From: Felix Manlunas <felix.manlunas@cavium.com>
Date: Tue, 16 May 2017 11:28:00 -0700
Subject: liquidio: fix PF falsely indicating success at setting MAC address of
 a nonexistent VF
Patch-mainline: v4.13-rc1
Git-commit: 0d9a5997842756f859032ae3efcaf79715a51883
References: bsc#1067126

In the function assigned to .ndo_set_vf_mac, check the validity of the
vfidx argument before proceeding to tell the firmware to set the VF MAC
address.

Signed-off-by: Felix Manlunas <felix.manlunas@cavium.com>
Signed-off-by: Derek Chickles <derek.chickles@cavium.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/cavium/liquidio/lio_main.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/cavium/liquidio/lio_main.c
+++ b/drivers/net/ethernet/cavium/liquidio/lio_main.c
@@ -3698,6 +3698,9 @@ static int liquidio_set_vf_mac(struct ne
 	struct octeon_device *oct = lio->oct_dev;
 	int retval;
 
+	if (vfidx < 0 || vfidx >= oct->sriov_info.num_vfs_alloced)
+		return -EINVAL;
+
 	retval = __liquidio_set_vf_mac(netdev, vfidx, mac, true);
 	if (!retval)
 		cn23xx_tell_vf_its_macaddr_changed(oct, vfidx, mac);
